core/src/main/scala/org/apache/spark/util/ShutdownHookManager.scala

```diff
@@ -209,9 +209,7 @@ private [util] class SparkShutdownHookManager {
 private class SparkShutdownHook(private val priority: Int, hook: () => Unit)
   extends Comparable[SparkShutdownHook] {
 
-  override def compareTo(other: SparkShutdownHook): Int = {
-    other.priority - priority
-  }
+  override def compareTo(other: SparkShutdownHook): Int = Integer.compare(other.priority, priority)
```

Comment:
why don't you just call `other.priority.compareTo(priority)`?

Reply:
no reason

Reply:
Either is fine. Integer.compare() avoids boxing but it won't matter at all here